test/scenarios/cloudnative/specific_agent_version/specific_agent_version.go

```diff
@@ -79,10 +66,13 @@ func getAvailableVersions(secret tenant.Secret, t *testing.T) []string {
 	return versions
 }
 
-func assessVersionChecks(builder *features.FeatureBuilder, version version.SemanticVersion, sampleApp sample.App) {
-	builder.Assess("restart sample apps", sampleApp.Restart)
-	cloudnative.AssessSampleInitContainers(builder, sampleApp)
-	builder.Assess("check env vars of init container", checkVersionInSampleApp(version, sampleApp))
+func assessVersionChecks(testDynakube dynatracev1beta1.DynaKube) features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		daemonset, err := oneagent.Get(ctx, environmentConfig.Client().Resources(), testDynakube)
+		require.NoError(t, err)
+		require.Contains(t, daemonset.Spec.Template.Spec.Containers[0].Image, testDynakube.OneAgentVersion())
```

Comment:
does `testDynakube.OneAgentVersion()` return image name? (for instance `quay.io/dynatrace-operator:v0.10.4` or is it `v0.10.4`). The function name is misleading if it is image name.

Reply:
We check if the string **contains** the version, so it should be fine. 
The OneAgentVersion returns the version without the image path as you said.